File: src/api/progress.ts
import { invoke } from "@tauri-apps/api/core";
import type { ProgressInfo } from "@/types";

export async function getGoalProgress(goalId: string): Promise<ProgressInfo> {
  return invoke("get_goal_progress", { goalId });
}

export async function getAllGoalsProgress(): Promise<ProgressInfo[]> {
  return invoke("get_all_goals_progress");
}

File: src-tauri/src/commands/progress.rs
use tauri::State;

use crate::db::models::ProgressInfo;
use crate::db::DbPool;
use crate::error::AppResult;
use crate::services::progress_service;

/// 获取单个目标的进度
#[tauri::command]
pub async fn get_goal_progress(
    goal_id: String,
    state: State<'_, DbPool>,
) -> AppResult<ProgressInfo> {
    progress_service::calc_goal_progress(&state.0, &goal_id).await
}

/// 获取所有目标的进度
#[tauri::command]
pub async fn get_all_goals_progress(state: State<'_, DbPool>) -> AppResult<Vec<ProgressInfo>> {
    progress_service::calc_all_goals_progress(&state.0).await
}
